File: app/schemas/memory.py
from pydantic import Field

from app.schemas.base import CamelModel
from app.schemas.common import TimestampedSchema


class LongTermMemoryBase(CamelModel):
    conversation_id: str | None = None
    persona_id: str | None = None
    memory_scope: str = "persona"
    content: str
    tags: list[str] = Field(default_factory=list)
    metadata: dict = Field(default_factory=dict, validation_alias="metadata_")


class LongTermMemoryCreate(LongTermMemoryBase):
    pass


class LongTermMemoryRead(LongTermMemoryBase, TimestampedSchema):
    id: str
    vector_id: str | None = None


class MemorySearchRequest(CamelModel):
    query: str
    conversation_id: str | None = None
    persona_id: str | None = None
    memory_scope: str | None = None
    tags: list[str] | None = None
    limit: int = 5


class MemorySummarizeRequest(CamelModel):
    conversation_id: str
    force: bool = False


class MemorySummaryRead(TimestampedSchema):
    id: str
    conversation_id: str
    summary: str
    source_message_count: int

File: src/types/memory.ts
export interface LongTermMemory {
  id: string
  conversationId?: string | null
  personaId?: string | null
  memoryScope: string
  content: string
  tags: string[]
  metadata: Record<string, unknown>
  vectorId?: string | null
  createdAt: string
  updatedAt: string
}

export interface LongTermMemoryCreateInput {
  conversationId?: string | null
  personaId?: string | null
  memoryScope: string
  content: string
  tags: string[]
  metadata?: Record<string, unknown>
}

export interface MemorySearchInput {
  query: string
  conversationId?: string | null
  personaId?: string | null
  memoryScope?: string | null
  tags?: string[] | null
  limit?: number
}

export interface MemorySummaryResult {
  id: string
  summary: string
  sourceMessageCount: number
}
